### PR TITLE
huc6270.cpp: Use device_memory_interface for VRAM interface

### PR DESCRIPTION
**Pull request by cam900**
Implement 4 color background mode, Fix spacing, Add hardware notes
Add input clocks for HuC6270s, Reduce defines for clocks
battlera.cpp: Fix clock values related on-board XTAL
pce.h: Remove unused tag define

### CODE REVIEW

File: src/devices/video/huc6270.h [hunk 1]
```diff
@@ -4,6 +4,67 @@
...
+    /HSYN: Horizontal sync
+    /VSYN: Vertical sync
+    MA0-15: Memory address
+    MD0-15: Memory data
+    VD0-7: Pixel data output
+    SP8G: Sprite/Background flag, Also highest bit of pixel data output
+    A0-1: Host interface address
+    D0-15: Host interface data
+    8/16: 8 or 16 bit host interface mode?
+    DISP: Unknown, display related?
```
Comment (angelosa) [inline src/devices/video/huc6270.h:66]: According to HuC6270 manual DISP pin is programmable and tied to video sync signals, presumably having sense in external genlocking cfr. page _H7-6_.

File: src/mame/drivers/paranoia.cpp [hunk 2]
```diff
@@ -92,6 +93,11 @@ void paranoia_state::pce_io(address_map &map)
 	map(0x00, 0x03).rw("huc6270", FUNC(huc6270_device::read), FUNC(huc6270_device::write));
 }
 
+void paranoia_state::vdc_mem(address_map &map)
```
Comment (angelosa) [inline src/mame/drivers/paranoia.cpp:96]: At this point `vdc_mem` can totally be derived from `pce_common_state` as protected function, no need to repeat memory maps in clients.

**Comment by galibert** [comment]
What's the point of plonking an address_space in the middle when it's
always ram and nothing else, apart from slowing things down?

  OG.

**Comment by cam900** [comment]
> What's the point of plonking an address_space in the middle when it's always ram and nothing else, apart from slowing things down?

for allow variable configuration cases(bankswitched, unmapped area or mirroring).

**Comment by MooglyGuy** [comment]
So cite some variable configuration cases or stop making changes just for the sake of making them.

This is a consistent problem with you, you make changes with no perceptible benefit for either the machine or for the development process. You need to stop doing this or your changes are going to start getting rejected more broadly.

**Comment by cam900** [comment]
I Close this for adjust later.

**Comment by cuavas** [comment]
This is the kind of thing where MAME has historically encouraged doing things in a way that’s more representative of the hardware if possible.  This chip does just have an external memory interface, so hard-coding for RAM in the device is an over-simplification.  (That said, I probably wouldn’t have gone out of my way to change it for this device unless I came across a system using this chip with banked, mirrored or non-contiguous video RAM.)

There are also side-benefits to using an address space, like making it more accessible to cheats and Lua scripts.

How much performance impact does this have on any systems actually using the chip?  Are there any cases where the VRAM access patterns mean this is enough overhead to cause a problem?

**Comment by angelosa** [comment]
In theory the unmapped portion should be mapped as open bus, which can be significantly different between vanilla PCE/battle rangers/supergraphx, need to test that in HW before having a definitve answer tho (may really give back the last accessed video property value instead).

I also have plans in having the VRAM to be dumped in a debugger command (if there isn’t a generic alias already) which of course should be mapped at driver level instead of device if we decide to externalize it out.

**Comment by angelosa** [comment]
Forgot to mention: current main bottleneck with PCE performance is the H/VSYNC signal stuff given that they synchronize with HuC6260, I have no idea about how to possibly improve given the full picture.

**Pull request closed without merging**